### PR TITLE
basic Fortran interface

### LINKED ISSUE #30: add fortran interface

### PR DESCRIPTION
**Pull request by q-posev**
consistent with C backend, initial wrapping implemented without C-related types like c_ptr, c_int etc.
fixes #30

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -35,19 +38,24 @@ endif
...
 libtrexio.so: $(OBJECT_FILES) $(HEADER_FILES)
 	$(CC) -shared $(OBJECT_FILES) -o libtrexio.so
 
-test: libtrexio.so test.c
+fortran: libtrexio.so trexio_f.f90
+	$(FC) $(FFLAGS) -c trexio_f.f90 -o trexio_f.o
+
+test: libtrexio.so test.c test.f90 trexio_f.o
+	rm -f -r test_text_fort/
 	$(CC) $(CFLAGS) $(INCLUDE) -Wl,-rpath,$(PWD) -L. test.c -ltrexio $(LIBS) -o test
```
Comment (scemama) [inline src/Makefile:53]:
```suggestion
test_c: libtrexio.so test.c
	$(CC) $(CFLAGS) $(INCLUDE) -Wl,-rpath,$(PWD) -L. test.c -ltrexio $(LIBS) -o test

test_f: libtrexio.so test.f90 trexio_f.o
	$(CC) $(CFLAGS) $(INCLUDE) -Wl,-rpath,$(PWD) -L. test.f90 trexio_f.o -ltrexio $(LIBS) -o test_f

test: test test_f
	$(RM) -r test_text
	$(RM) -r test_text_fort
	./test
	./test_f
```

File: src/Makefile [hunk 2]
```diff
@@ -35,19 +38,24 @@ endif
...
+	$(CC) $(CFLAGS) $(INCLUDE) -Wl,-rpath,$(PWD) -L. test.f90 trexio_f.o -ltrexio $(LIBS) -o test_f
 
 clean:
-	rm -f *.o libtrexio.so test test_*.h5 
-	rm -r trexio_test/
+	rm -f *.o libtrexio.so test test_*.h5 test_f
+	rm -r test_text_fort/
```
Comment (scemama) [inline src/Makefile:58]:
```suggestion
	$(RM) *.o libtrexio.so test test_*.h5 test_f
	$(RM) -r test_text_fort/
```

File: src/templates_front/build.sh [hunk 3]
```diff
@@ -2,14 +2,21 @@
 
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
 
+# prefixes
 cat $DIR/prefix_front.c   > trexio.c
 cat $DIR/prefix_front.h   > trexio.h
 cat $DIR/prefix_s_front.h > trexio_s.h
+cat $DIR/prefix_fortran.f90   > trexio_f.f90
 
+# c front end
```
Comment (scemama) [inline src/templates_front/build.sh:11]: `DIR` might contain spaces. You should use `"$DIR"` instead when you use it.

**Review by scemama — APPROVED** [review]

**Comment by scemama** [comment]
Very nice!

File: src/Makefile (hunk 2 above)
Comment (q-posev) [reply src/Makefile:58]: I don't see the definiton of $(RM) in Makefile. Should it be something like "rm -f" ?

**Comment by scemama** [comment]
By default, $(RM) is rm -f in make. It is better to use the variables because on
some systems you might want to change the command to remove files.
Check this out :
https://www.gnu.org/software/make/manual/html_node/Implicit-Variables.html

Evgeny Posenitskiy writes:

> @q-posev commented on this pull request.
>
>
>
>> +	rm -f *.o libtrexio.so test test_*.h5 test_f
> +	rm -r test_text_fort/
>
> I don't see the definiton of $(RM) in Makefile. Should it be something like "rm -f" ?


-- 
Anthony Scemama
IRSAMC / Laboratoire de Chimie et Physique Quantiques
UMR5626 CNRS/UPS
Toulouse

**Pull request merged**